react/src/components/Widget/WidgetContainer.tsx

```diff
@@ -94,12 +92,18 @@ export const WidgetContainer: React.FC<WidgetContainerProps> = withSnackbar(
 
     const addressDetails = useAddressDetails(address, active && !success);
 
-    const isFiat: boolean = currency !== 'BCH' && currency !== 'XEC';
-
     const getPrice = useCallback(async (): Promise<void> => {
       try {
-        if (isFiat) {
-          const data = await getFiatPrice(currency);
+        if (isFiat(currency) && isValidCashAddress(address)) {
+          const data = await getBchFiatPrice(currency);
+
+          const { price } = data;
+          setLoading(false);
+          setPrice(price);
+        } else if (isFiat(currency) && isValidXecAddress(address)) {
+          // TODO: THIS SHOULD POINT TO THE XEC FUNCTION. THIS IS TEMPORARY WHILE WAITING FOR BACKEND
```

Comment:
Is this done? Looks like you're using an XEC function

Reply:
oops, forgot to remove